wire/service.go

```diff
@@ -0,0 +1,155 @@
+	resp, err := makeWhatsMyURIResponse(s.Address()).Message()
+	if err != nil {
+		writeErrorResponse(ctx, w, errors.Wrap(err, "could not make response"), WhatsMyURICommand)
+		return
+	}
+	writeResponse(ctx, w, resp, WhatsMyURICommand)
+}
+
+func (s *service) buildInfo(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	// resp := birch.NewDocument(birch.EC.String("version", "0.0.0"))
+	// writeSuccessResponse(w, resp, BuildInfoCommand)
+	resp, err := makeBuildInfoResponse("0.0.0").Message()
+	if err != nil {
+		writeErrorResponse(ctx, w, errors.Wrap(err, "could not make response"), BuildInfoCommand)
+		return
+	}
+	writeResponse(ctx, w, resp, BuildInfoCommand)
+}
+
+func (s *service) getLog(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	resp, err := makeGetLogResponse([]string{}).Message()
+	if err != nil {
+		writeErrorResponse(ctx, w, errors.Wrap(err, "could not make response"), GetLogCommand)
+		return
+	}
+	writeResponse(ctx, w, resp, GetLogCommand)
+}
+
+func (s *service) getFreeMonitoringStatus(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	writeNotOKResponse(ctx, w, GetFreeMonitoringStatusCommand)
+}
+
+func (s *service) replSetGetStatus(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	writeNotOKResponse(ctx, w, ReplSetGetStatusCommand)
+}
+
+func (s *service) listCollections(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	writeNotOKResponse(ctx, w, ListCollectionsCommand)
+}
+
+func (s *service) registerHandlers() error {
+	for name, handler := range map[string]mrpc.HandlerFunc{
+		// Required initialization commands
+		IsMasterCommand:                s.isMaster,
+		WhatsMyURICommand:              s.whatsMyURI,
+		BuildinfoCommand:               s.buildInfo,
+		BuildInfoCommand:               s.buildInfo,
+		GetLogCommand:                  s.getLog,
+		ReplSetGetStatusCommand:        s.replSetGetStatus,
+		GetFreeMonitoringStatusCommand: s.getFreeMonitoringStatus,
+		ListCollectionsCommand:         s.listCollections,
+
+		// Manager commands
+		ManagerIDCommand:     s.managerID,
+		CreateProcessCommand: s.managerCreateProcess,
+		ListCommand:          s.managerList,
+		GroupCommand:         s.managerGroup,
+		GetCommand:           s.managerGet,
+		ClearCommand:         s.managerClear,
+		CloseCommand:         s.managerClose,
+
+		// Process commands
+		InfoCommand:                    s.processInfo,
+		RunningCommand:                 s.processRunning,
+		CompleteCommand:                s.processComplete,
+		WaitCommand:                    s.processWait,
+		SignalCommand:                  s.processSignal,
+		RegisterSignalTriggerIDCommand: s.processRegisterSignalTriggerID,
+		RespawnCommand:                 s.processRespawn,
+		TagCommand:                     s.processTag,
+		GetTagsCommand:                 s.processGetTags,
+		ResetTagsCommand:               s.processResetTags,
+	} {
+		if err := s.RegisterOperation(&mongowire.OpScope{
```

Comment:
while I think we definitely need these kinds of handler funcs in this option, should we open a ticket for mrpc that gives us a different interface for registering handlers that's more like gimlet's (rest/v2)'s framework that would be easier to test?

Reply:
I can open a ticket but don't know what different interface you have in mind. As far as I'm aware, the gimlet method of adding routes is basically the same thing.